### PR TITLE
materialDesign attribute for Material Design Charts

### PR DESCRIPTION
**Pull request by SmokyBob**
#30 implementation for the charts that have a Material Design Beta version

### CODE REVIEW

File: google-chart.html [hunk 1]
```diff
@@ -179,6 +179,22 @@
         selection: [],
 
         chartTypes: null,
+        /**
+         * Try to use the Material Design version of the chart.
+         *
+         * Material Design Charts are avaliable in BETA for the following types:
+         * <ul>
```
Comment (ebidel) [inline google-chart.html]: You can use markdown in the docstring.

File: google-chart.html [hunk 2]
```diff
@@ -263,6 +279,12 @@
                           { selection: this.chartObject.getSelection() });
                   }.bind(this));
 
+              if (this.materialDesign){
```
Comment (ebidel) [inline google-chart.html]: spacing: `if (this.materialDesign){` -> `if (this.materialDesign) {`

File: google-chart.html [hunk 3]
```diff
@@ -263,6 +279,12 @@
...
+                if (this.type=='bar'){
```
Comment (ebidel) [inline google-chart.html]: same

File: google-chart.html [hunk 4]
```diff
@@ -263,6 +279,12 @@
...
+                  this.options.bars='horizontal';
```
Comment (ebidel) [inline google-chart.html]: spaces around `=`

File: google-chart.html [hunk 5]
```diff
@@ -279,39 +301,64 @@
         loadChartTypes: function() {
           this.chartTypes = {
             'area': google.visualization.AreaChart,
-            'bar': google.visualization.BarChart,
+            //'bar': google.visualization.BarChart,
```
Comment (ebidel) [inline google-chart.html]: why are these commented out?

File: google-chart.html [hunk 6]
```diff
@@ -279,39 +301,64 @@
...
-            'scatter': google.visualization.ScatterChart,
+            //'scatter': google.visualization.ScatterChart,
             'stepped-area': google.visualization.SteppedAreaChart,
             'table': google.visualization.Table,
             'gauge': google.visualization.Gauge
           };
+
+          if (this.materialDesign){
+            this.chartTypes['bar']= google.charts.Bar;
+            this.chartTypes['column']= google.charts.Bar;
```
Comment (ebidel) [inline google-chart.html]: spacing needs to be fixed throughout. Please see https://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml

File: google-chart.html [hunk 7]
```diff
@@ -263,6 +279,12 @@
                           { selection: this.chartObject.getSelection() });
                   }.bind(this));
 
+              if (this.materialDesign){
+                if (this.type=='bar'){
+                  this.options.bars='horizontal';
+                }
+                this.options=google.charts.Bar.convertOptions(this.options);
```
Comment (ebidel) [inline google-chart.html]: what about the Column case? `google.charts.Column.convertOptions(options)`

**Comment by ebidel** [comment]
Reviewed

File: google-chart.html (hunk 7 above)
Comment (SmokyBob) [inline google-chart.html]: My bad, followed the sample in the [documentation](https://developers.google.com/chart/interactive/docs/gallery/columnchart#Material) which is using the `google.charts.Bar` object but a bit later is suggested to use `google.charts.Column`.

File: google-chart.html (hunk 5 above)
Comment (SmokyBob) [inline google-chart.html]: Bad habits, removing them in following commits

**Comment by SmokyBob** [comment]
added commits for the fixes.
Not 100% sure about the doc string.
I've used markdown for the list, but not for the links for consistency with the links in the docstring, but I've noticed that other google web components are like `google-youtube-upload` and `google-sheets` use markdown for the link too.

Is markdown for the link the preferred way to express them in the docstring?

File: demo.html [hunk 8]
```diff
@@ -184,6 +184,16 @@
...
   </google-chart>
 
+  <p>Here's a Material Design bar chart:</p>
+
+  <google-chart
+    type='bar'
+    options='{"title": "Days in a month"}'
+    cols='[{"label": "Month", "type": "string"},{"label": "Days", "type": "number"}]'
+    rows='[["Jan", 31],["Feb", 28],["Mar", 31],["Apr", 30],["May", 31],["Jun", 30]]'
+    materialDesign>
```
Comment (ebidel) [inline demo.html]: `material-design`

File: demo.html [hunk 9]
```diff
@@ -224,6 +234,16 @@
...
   </google-chart>
 
+  <p>Here's a Material Design column chart:</p>
+
+  <google-chart
+    type='column'
+    options='{"title": "Days in a month"}'
+    cols='[{"label": "Month", "type": "string"},{"label": "Days", "type": "number"}]'
+    rows='[["Jan", 31],["Feb", 28],["Mar", 31],["Apr", 30],["May", 31],["Jun", 30]]'
+    materialDesign>
```
Comment (ebidel) [inline demo.html]: `material-design`

File: demo.html [hunk 10]
```diff
@@ -268,6 +288,16 @@
...
   </google-chart>
 
+  <p>Here's a Material Design line chart:</p>
+
+  <google-chart
+    type='line'
+    options='{"title": "Days in a month"}'
+    cols='[{"label": "Month", "type": "string"},{"label": "Days", "type": "number"}]'
+    rows='[["Jan", 31],["Feb", 28],["Mar", 31],["Apr", 30],["May", 31],["Jun", 30]]'
+    materialDesign>
```
Comment (ebidel) [inline demo.html]: same

File: demo.html [hunk 11]
```diff
@@ -290,6 +320,20 @@
...
+  <google-chart
+    type='scatter'
+    options='{"legend": "none"}'
+    data='[["A", "B"],
+           [20, 45],
+           [31, 66],
+           [50, 80],
+           [77, 50],
+           [68, 15]]'
+    materialDesign>
```
Comment (ebidel) [inline demo.html]: same

**Comment by ebidel** [comment]
This PR is pre 1.0. Can you update it? Otherwise, we'll close.

**Comment by SmokyBob** [comment]
Sorry not going to be able to update the PR for at least a couple weeks.

Feel free to close the PR, maybe I'll re implement the Issue later on the latest version

**Comment by alanent** [comment]
Someone can help me for the integration in angular ? https://stackoverflow.com/q/51153920/9962015

**Pull request closed without merging**